Pass file contents to linter via STDIN

jamieconnolly opened the pull request:
With the release of scss-lint v0.43.0, you can now lint files via STDIN. This PR passes the file contents directly to scss-lint, rather than generating a temp file.

Is there anything needed doing to let users know that v0.43.0 onwards is required?

Fixes #103

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/atomlinter/linter-scss-lint/136)
<!-- Reviewable:end -->

jamieconnolly commented:
The build is failing on a peer dependency conflict between babel-eslint@5.0.4 (requires eslint < 2.3.0) and eslint-config-airbnb@6.2.0 (requires eslint ^2.4.0). The solution is either upgrading to babel-eslint 6.x (see #135) or by keeping eslint-config-airbnb at 6.1.0 (requires eslint ^ 2.2.0, but that would need to change as babel-eslint requires eslint < 2.3.0).

Arcanemagus commented:
I'll get the dependency issues cleared up in a few minutes.

Arcanemagus commented:
As for the version.... we can either:

* Make it an option to use the temp file method for anyone that can't update their linter for some reason
* Auto-detect the version and use the appropriate method (slow for first lint to run detection)
* Only work with the current version

I'd say either of the latter two options are acceptable, although we should do a major version bump if we do the last.

Arcanemagus commented:
Wow, this has needed some love for a while:
![image](https://cloud.githubusercontent.com/assets/427137/14213231/babae2f0-f7eb-11e5-935a-cb6f7c3f932a.png)

@jamieconnolly assuming @SpainTrain has no issues, would you be interested in becoming a maintainer here?

jamieconnolly commented:
Thanks for sorting out the dependency issue @Arcanemagus, I've now rebased the branch and the tests pass.

I've also fixed an issue where SCSS Lint expects a relative file path (to the location of the config file) as the `--stdin-file-path` option. For the times when there is no local config file, I've made it default to a path that's relative to the project directory. Does this make sense, or should I change it to be the absolute path?

I have no issues with helping maintain this project if you need it.

jamieconnolly commented:
Also, with the versioning, my choice would be to only use the current version and make a major version bump. Something should get put in the README.md about requiring scss-lint v0.43.0 as a minimum too.

If needed, we could also do something clever with the errors. Depending on what get's thrown back by SCSS Lint pre v0.43.0, we could forcibly error out by letting the user know that they need to upgrade their version. I can rummage through existing linter plugins to see if there's any precedent with versioning if we need? Or should that be only done if we start getting people reporting issues?

Arcanemagus commented:
It's pretty much guaranteed that no matter what we do there will be a user that doesn't bother reading _anything_ and simply reports "it's broken suddenly!" :stuck_out_tongue: :disappointed:

I'm not sure how to answer the absolute/relative question as I don't know how this linter works... use whichever method causes it to work properly?

It might be nice to see what older versions throw out when fed input like this and check for that error text and show an error to the user. Linter plugins right now don't handle ancient versions of their linter that well, some do okay (warning if old version is found), most just don't even bother though.

Arcanemagus commented:
Invite to the team maintaining this sent :wink:

If you have any questions feel free to hop in the #linter-plugin-dev channel on [Atom's Slack](http://atom-slack.herokuapp.com/).

Releasing new versions is done through `apm` by running `apm publish {major|minor|patch}` (like `npm`).

Arcanemagus commented:
@jamieconnolly You can merge your own PR now btw :stuck_out_tongue:.

Arcanemagus commented:
Hooray for signed commits :+1: :stuck_out_tongue:

Just bothered to figure out how to do a signed release myself:

```ShellSession
<commit everything, including the version number change that apm usually does>
git tag -s vx.y.z
git push --follow-tag
apm publish --tag v4.0.0
```

jamieconnolly commented:
Nice, that saves me working out how do it through APM. Many thanks!
